### PR TITLE
feat: check for unsupported sierra version in stateless validatons

### PR DESCRIPTION
**Pull request by ArniStarkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)

### CODE REVIEW

**Comment by ArniStarkware** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/247?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#247** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/247?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#226** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/226?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#244](https://github.com/starkware-libs/mempool/pull/244) <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/244?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#225** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/225?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ArniStarkware and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ArniStarkware** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/247?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#247** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/247?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#226** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/226?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#244](https://github.com/starkware-libs/mempool/pull/244) <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/244?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#225** <a href="https://app.graphite.dev/github/pr/starkware-libs/mempool/225?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ArniStarkware and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Yael-Starkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 6 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: all files reviewed, 3 unresolved discussions (waiting on @ArniStarkware and @dafnamatsry)

___
*[`crates/gateway/src/errors.rs` line 55 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kPm18CpD1EQDtsyRr:-O-kPm18CpD1EQDtsyRs:bci11ez) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/errors.rs#L55)):*
> ```XML
>     #[error(
>         "Compiled versions older than {min_version} or newer than {max_version} are not \
>          supported. Got {version}."
> ```

no sure I understand this msg, what does it mean "Got version" ? maybe rephrase.



_Suggestion:_
```XML
Sierra version {version}  is older....
```
___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 133 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kQck18tHdjB-gec0p:-O-kQck18tHdjB-gec0q:bh7f4a0) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator.rs#L133)):*
> ```XML
>         // Check that the version is lower than the latest version allowing higher patch versions
>         // (i.e. we ignore the Z part in a version X.Y.Z).
>         let max_minor_sierra_version = VersionId { patch: 0, ..max_sierra_version };
> ```

iiuc the config of max sierra version should always have patch = 0 ? maybe we should enforce it ?
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 257 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kRTjm7wHB7gnF0w3T:-O-kRTjm7wHB7gnF0w3U:bw6m1jg) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator_test.rs#L257)):*
> ```XML
> #[case::valid_sierra_version(VersionId { major: 1, minor: 3, patch: 0 })]
> #[case::max_sierra_version(MAX_SIERRA_VERSION)]
> fn positive_flow_test_declare_sierra_version(#[case] sierra_version: VersionId) {
> ```

lets be consistent in both tests and define this input with the same type (as either VersionId or Vec<StarkFelt> )



_Code quote:_
```XML
sierra_version: VersionId
```

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: all files reviewed, 3 unresolved discussions (waiting on @dafnamatsry and @Yael-Starkware)

___
*[`crates/gateway/src/errors.rs` line 55 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kPm18CpD1EQDtsyRr:-O-kgh8l67egKkfPcH9n:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/errors.rs#L55)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

no sure I understand this msg, what does it mean "Got version" ? maybe rephrase.
</blockquote></details>

Done.
___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 133 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kQck18tHdjB-gec0p:-O-khfMYBwZsODdzaEb-:b3aepb) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator.rs#L133)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

iiuc the config of max sierra version should always have patch = 0 ? maybe we should enforce it ?
</blockquote></details>

You are correct, but this validation is not worth our time.
The `max_sierra_version` is only a part of the config.
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 257 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kRTjm7wHB7gnF0w3T:-O-kq8WPBgq8F903MDNT:b25b5lp) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator_test.rs#L257)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

lets be consistent in both tests and define this input with the same type (as either VersionId or Vec<StarkFelt> )
</blockquote></details>

Done. Updated #248.

**Review by Yael-Starkware — APPROVED** [review]
> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: all files reviewed, 1 unresolved discussion (waiting on @ArniStarkware and @dafnamatsry)

___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 133 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kQck18tHdjB-gec0p:-O-oK6ImCR2uh0IN44-I:btdj574) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator.rs#L133)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

You are correct, but this validation is not worth our time.
The `max_sierra_version` is only a part of the config.
</blockquote></details>

so maybe assert somewhere earlier when we read the config? to make sure it is a valid value?

**Review by Yael-Starkware — APPROVED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: all files reviewed, 1 unresolved discussion (waiting on @ArniStarkware and @dafnamatsry)

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: 2 of 7 files reviewed, 1 unresolved discussion (waiting on @dafnamatsry and @Yael-Starkware)

___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 133 at r2](https://reviewable.io/reviews/starkware-libs/mempool/247#-O-kQck18tHdjB-gec0p:-O0A9FoE6t89l5YX4R72:b6p1bna) ([raw file](https://github.com/starkware-libs/mempool/blob/120d92de8d097a606672c004b6fe3716020f125a/crates/gateway/src/stateless_transaction_validator.rs#L133)):*
<details><summary><i>Previously, Yael-Starkware (YaelD) wrote…</i></summary><blockquote>

so maybe assert somewhere earlier when we read the config? to make sure it is a valid value?
</blockquote></details>

Now the config is validated.

**Review by dafnamatsry — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r4, 4 of 6 files at r5.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: 5 of 7 files reviewed, 4 unresolved discussions (waiting on @ArniStarkware and @Yael-Starkware)

___
*[`crates/gateway/src/compiler_version.rs` line 83 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0D_ASxCZlGCWGDXvq8:-O0D_ASy7HmehHESIez4:b-mc9y5l) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/compiler_version.rs#L83)):*
> ```XML
>             StarkFelt::from(u64::try_from(self.patch).unwrap()),
>         ]
>     }
> ```

Please put it in some test utils, and rename to `create_sierra_program(version: &VersionId)`.



_Code quote:_
```XML
    pub fn into_sierra_program(&self) -> Vec<StarkFelt> {
        vec![
            StarkFelt::from(u64::try_from(self.major).unwrap()),
            StarkFelt::from(u64::try_from(self.minor).unwrap()),
            StarkFelt::from(u64::try_from(self.patch).unwrap()),
        ]
    }
```
___
*[`crates/gateway/src/config.rs` line 71 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0Dgg6d5ceQcdPZfWEd:-O0Dgg6d5ceQcdPZfWEe:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/config.rs#L71)):*
> ```XML
> 
> #[derive(Clone, Debug, Serialize, Deserialize, Validate, PartialEq)]
> #[validate(schema(function = "validate_max_sierra_version"))]
> ```





_Suggestion:_
```XML
#[validate(schema(function = "validate_stateless_transaction_validator_config"))]
```
___
*[`crates/gateway/src/config_test.rs` line 31 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0DiAlr00KKpRQUe7PB:-O0DiAlr00KKpRQUe7PC:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/config_test.rs#L31)):*
> ```XML
>             )
>         )
>     });
> ```





_Suggestion:_
```XML
    assert!(config.validate().is_err());
```

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: 2 of 9 files reviewed, 4 unresolved discussions (waiting on @dafnamatsry and @Yael-Starkware)

___
*[`crates/gateway/src/compiler_version.rs` line 83 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0D_ASxCZlGCWGDXvq8:-O0Dq98OB8VguHVXFiOj:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/compiler_version.rs#L83)):*
<details><summary><i>Previously, dafnamatsry wrote…</i></summary><blockquote>

Please put it in some test utils, and rename to `create_sierra_program(version: &VersionId)`.
</blockquote></details>

Done.
___
*[`crates/gateway/src/config.rs` line 71 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0Dgg6d5ceQcdPZfWEd:-O0Dpb2x4zE8lsUO2yoO:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/config.rs#L71)):*
> ```XML
> 
> #[derive(Clone, Debug, Serialize, Deserialize, Validate, PartialEq)]
> #[validate(schema(function = "validate_max_sierra_version"))]
> ```

Done.
___
*[`crates/gateway/src/config_test.rs` line 31 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0DiAlr00KKpRQUe7PB:-O0Drh6R5ccO326Ojlq0:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/config_test.rs#L31)):*
> ```XML
>             )
>         )
>     });
> ```

Done.

**Review by dafnamatsry — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r5, 6 of 6 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: all files reviewed, 7 unresolved discussions (waiting on @ArniStarkware and @Yael-Starkware)

___
*[`crates/gateway/src/config.rs` line 142 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E80jIEeOdOHMIrUN9:-O0E80jIEeOdOHMIrUNA:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/config.rs#L142)):*
> ```XML
> }
> 
> /// In the max_sierra_version, the patch version should always be 0.
> ```


___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 120 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E8oKYDq9XmdooBvCH:-O0E8oKYDq9XmdooBvCI:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator.rs#L120)):*
> ```XML
>         let min_sierra_version = self.config.min_sierra_version;
>         let max_sierra_version = self.config.max_sierra_version;
>         // Check that the version is not too old.
> ```





_Suggestion:_
```XML
        let max_sierra_version = self.config.max_sierra_version;
        
        // Check that the version is not too old.
```
___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 138 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E93fSADdRKd3alEus:-O0E93fSADdRKd3alEut:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator.rs#L138)):*
> ```XML
>                 max_version: max_sierra_version,
>             });
>         }
> ```





_Suggestion:_
```XML

        // Check that the version is lower than the latest version allowing higher patch versions
        // (i.e. we ignore the Z part in a version X.Y.Z).
        let minor_sierra_version = VersionId { patch: 0, ..sierra_version };

        if sierra_version < min_sierra_version || max_sierra_version < minor_sierra_version {
            return Err(StatelessTransactionValidatorError::UnsupportedSierraVersion {
                version: sierra_version,
                min_version: min_sierra_version,
                max_version: max_sierra_version,
            });
        }
```
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 251 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EBHZOBu7jZMr2ue9G:-O0EBHZOBu7jZMr2ue9H:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/stateless_transaction_validator_test.rs#L251)):*
> ```XML
>     })
> ]
> fn test_declare_sierra_version(
> ```





_Suggestion:_
```XML
test_declare_sierra_version_failure
```
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 268 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EBYJz1thqhR3I2mfc:-O0EBYJz1thqhR3I2mfd:b-l2nc3) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator_test.rs#L268)):*
> ```XML
> #[case::min_sierra_version(create_sierra_program(&MIN_SIERRA_VERSION))]
> #[case::valid_sierra_version(create_sierra_program(&VersionId { major: 1, minor: 3, patch: 0 }))]
> #[case::max_sierra_version(create_sierra_program(&MAX_SIERRA_VERSION))]
> ```

Please add a case for with higher patch version.
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 269 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EAKlC6BfePSbPQ_Eg:-O0EAKlC6BfePSbPQ_Eh:undefined) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator_test.rs#L269)):*
> ```XML
> #[case::valid_sierra_version(create_sierra_program(&VersionId { major: 1, minor: 3, patch: 0 }))]
> #[case::max_sierra_version(create_sierra_program(&MAX_SIERRA_VERSION))]
> fn positive_flow_test_declare_sierra_version(#[case] sierra_program: Vec<StarkFelt>) {
> ```





_Suggestion:_
```XML
test_declare_sierra_version_success
```

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: 6 of 9 files reviewed, 7 unresolved discussions (waiting on @dafnamatsry and @Yael-Starkware)

___
*[`crates/gateway/src/config.rs` line 142 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E80jIEeOdOHMIrUN9:-O0ECpT91X9IIrw2Rn7w:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/config.rs#L142)):*
> ```XML
> }
> 
> /// In the max_sierra_version, the patch version should always be 0.
> ```

Done.
___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 120 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E8oKYDq9XmdooBvCH:-O0EFpCVBsic4-Peu07O:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator.rs#L120)):*
> ```XML
>         let min_sierra_version = self.config.min_sierra_version;
>         let max_sierra_version = self.config.max_sierra_version;
>         // Check that the version is not too old.
> ```

Done.
___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 138 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E93fSADdRKd3alEus:-O0EDopW2nSmwraxVskM:bwaqydp) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator.rs#L138)):*
> ```XML
>                 max_version: max_sierra_version,
>             });
>         }
> ```

Done.
Also, something for improved reliability.
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 251 at r5](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EBHZOBu7jZMr2ue9G:-O0EFHO95XznfG_YfXTZ:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/24c4b7d03e555ce4a1d86dc75a48cd300e2c8205/crates/gateway/src/stateless_transaction_validator_test.rs#L251)):*
> ```XML
>     })
> ]
> fn test_declare_sierra_version(
> ```

Done.
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 268 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EBYJz1thqhR3I2mfc:-O0EFh0f8G3oPqvzWkpC:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator_test.rs#L268)):*
<details><summary><i>Previously, dafnamatsry wrote…</i></summary><blockquote>

Please add a case for with higher patch version.
</blockquote></details>

Done.
___
*[`crates/gateway/src/stateless_transaction_validator_test.rs` line 269 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0EAKlC6BfePSbPQ_Eg:-O0EFn5mCpsiEztp0z2W:b-896fix) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator_test.rs#L269)):*
> ```XML
> #[case::valid_sierra_version(create_sierra_program(&VersionId { major: 1, minor: 3, patch: 0 }))]
> #[case::max_sierra_version(create_sierra_program(&MAX_SIERRA_VERSION))]
> fn positive_flow_test_declare_sierra_version(#[case] sierra_program: Vec<StarkFelt>) {
> ```

Done.

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247)* status: 6 of 9 files reviewed, 7 unresolved discussions (waiting on @dafnamatsry and @Yael-Starkware)

___
*[`crates/gateway/src/stateless_transaction_validator.rs` line 138 at r6](https://reviewable.io/reviews/starkware-libs/mempool/247#-O0E93fSADdRKd3alEus:-O0EKOzuC9khX1ca-vYA:bl2ngmi) ([raw file](https://github.com/starkware-libs/mempool/blob/b14ab309cd1a2ca671ae951596e33f1313366b49/crates/gateway/src/stateless_transaction_validator.rs#L138)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Done.
Also, something for improved reliability.
</blockquote></details>

Simplified even further in #281.

**Review by dafnamatsry — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/247#-:-O0EPP0r2r8ri3NIioUe:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Yael-Starkware)
